### DISCUSSION TITLE
✈ 🔗 Release a Template Library API to allow bundling templates with themes (Related to #417) (Migrated)

### DISCUSSION THREAD

**Discussion by WPDevHQ** [elementor/elementor#30213] [Feature Request]
Will it be possible for the plugin to auto detect theme bundled custom templates?

**Comment by yehudah** [comment]
Sorry, can you explain ?
what the purpose ?

**Comment by WPDevHQ** [comment]
The purpose is to redistribute theme specific library templates.

The scenario is that I as a theme author create custom elementor templates that are specific for theme and in order to pass them on to my end user with the current setup, they a) have to download the theme, install and activate and then b) download the custom library templates and then import them in to the plugin.

What I'm proposing is to have those custom library templates bundled within a folder in the theme & upon theme installation and activation, those templates would automatically be detected and imported in to the "My Library" and thus be available for immediate use.

We could go further (more work) and not import the theme but the API would detect and make them available via a 3rd tab marked as "Theme Templates" in the Library pop up panel.

**Comment by yehudah** [comment]
"My Library" are just Custom Post Types so you can code something to import them from a json on theme activation.

**Comment by arielk** [comment]
just for the reference #417

**Comment by arbex5** [comment]
Awesome. This will be a great feature for elementor

**Comment by jgut12** [comment]
Any update on this issue? Thanks!

**Comment by nicholaszein** [comment]
Hi there,

### ✈️ This Feature Request has been migrated from issue #454 

if you still think this request should be considered for our roadmap, please cast your **⬆️ Upvote** on the first comment.

Don't worry, your comments, reactions, and participation in the old issue will be accounted for when considering this Feature Request for our Roadmap.

Happy voting! 🙂
